scala_proto/private/scalapb_aspect.bzl

```diff
@@ -0,0 +1,267 @@
+load(
+    "//scala/private:common.bzl",
+    "write_manifest_file",
+)
+load("//scala/private:rule_impls.bzl", "compile_scala")
+
+load("//scala_proto/private:proto_to_scala_src.bzl", "proto_to_scala_src")
+load("//scala_proto/private:dep_sets.bzl", "SCALAPB_DEPS", "GRPC_DEPS")
+
+
+ScalaPBAspectInfo = provider(fields = [
+    "proto_info",
+    "src_jars",
+    "output_files",
+    "java_info",
+])
+
+
+ScalaPBImport = provider(fields = [
+    "java_info",
+    "proto_info",
+])
+
+
+ScalaPBInfo = provider(fields = [
+    "aspect_info",
+])
+
+
+
+def lift_proto_to_struct(tis):
+    return struct(
+        direct_sources = tis.direct_sources,
+        transitive_sources = tis.transitive_sources,
+        # transitive = depset(transitive = [t.transitive for t in tis]),
+        # srcs = depset(transitive = [t.srcs for t in tis]),
+        # import_flags = [t.import_flags for t in tis],
+        # deps = depset(transitive = [t.deps for t in tis]),
+    )
+
+def merge_proto_infos(tis):
+    return struct(
+        transitive_sources = [t.transitive_sources for t in tis],
+        # transitive = depset(transitive = [t.transitive for t in tis]),
+        # srcs = depset(transitive = [t.srcs for t in tis]),
+        # import_flags = [t.import_flags for t in tis],
+        # deps = depset(transitive = [t.deps for t in tis]),
+    )
+
+
+def merge_scalapb_aspect_info(scalapbs):
+    return ScalaPBAspectInfo(
+        src_jars = depset(transitive = [s.src_jars for s in scalapbs]),
+        output_files = depset(transitive = [s.output_files for s in scalapbs]),
+        proto_info = merge_proto_infos([s.proto_info for s in scalapbs]),
+        java_info = java_common.merge([s.java_info for s in scalapbs]),
```

Comment:
did you see how this affects, if at all, on intellij? Not saying it should but just battle scarred.

Reply:
No I rarely enough use intellij, i'm happy to try out some steps on intellij if you know of any common failure modes/usual utility that should work here.